api/server.spec.js

```diff
@@ -25,17 +26,128 @@ describe('The API Server', () => {
     //   // write a beforeEach hook that will populate your test DB with data
     //   // each time this hook runs, you should save a document to your db
     //   // by saving the document you'll be able to use it in each of your `it` blocks
+    return Game.create({
+        title: 'California Games',
+        genre: 'Sports',
+        releaseDate: 'June 1987'
+      })
+      .then(game => {
+        gameId = game._id
+        // console.log(gameId)
+      })
   });
 
   afterEach(() => {
     //   // clear the games collection.
+    return Game.remove({})
   });
 
   it('runs the tests', () => {});
 
   // test the POST here
+  it("should post data to database", async () => {
+    await request(server)
+      .post('/api/games')
+      .send({
+        title: 'New York Games',
+        genre: 'Sports',
+        releaseDate: 'June 1982'
+      })
+      .expect(201)
+      .then(response => {
```

Comment:
So the cool thing about `async`/`await` is since we're telling JS that the code attached to the `await` keyword is a Promise, we _don't_ have to use `.then()` anymore. So we can omit it entirely and write cleaner code:

```js
it("should post data to database", async () => {
  const newGame = {
    title: 'New York Games',
    genre: 'Sports',
    releaseDate: 'June 1982'
  }
  const response = await request(server).post('/api/games').send(newGame).expect(201)
  expect(response.body).toHaveProperty('title', 'New York Games')
  expect(response.body).toHaveProperty('genre', 'Sports')
  expect(response.body).toHaveProperty('releaseDate', 'June 1982')
})
```